cmake fails when no yarp_devices are selected

**Describe the bug**
cmake fails when no "yarpmod" devices are selected
Error:

`` CMake Error at src/devices/CMakeLists.txt:93 (list):``                                                           
``list sub-command REMOVE_DUPLICATES requires list to be present.``

**To Reproduce**
pull the latest master
mkdir build && cd build
ccmake ..
set "OFF" to all devices with the prefix "ENABLE_yarpmod"
configure

**Expected behavior**
Cmake should still work without yarp devices    

**Configuration (please complete the following information):**
 - OS: ubuntu 16.04
 - yarp version: 3.0.0 (3.0.0+13-20180615.9+gitf23f86b) 
 - CMake: 3.5.1
-  compiler: GNU 5.4.0

**Additional context**
Google result here: https://cmake.org/pipermail/cmake/2009-January/026581.html
seems that it can be fixed by changing how the list variable is referenced.

--- a/doc/release/v3_0_1.md
+++ b/doc/release/v3_0_1.md
@@ -25,6 +25,8 @@ Bug Fixes
   be used instead).
 * Deprecation warnings when calling find_package(YARP) more than once are no
   longer printed.
+* Fix configuration when `YARP_COMPILE_CARRIER_PLUGINS` and
+  `YARP_COMPILE_DEVICE_PLUGINS` is set to OFF.
 
 
 ### Libraries

--- a/src/carriers/CMakeLists.txt
+++ b/src/carriers/CMakeLists.txt
@@ -34,7 +34,9 @@ install(TARGETS yarpcar
         ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
 include(YarpInstallBasicPackageFiles)
-list(REMOVE_DUPLICATES YARP_yarpcar_PRIVATE_DEPS)
+if(YARP_yarpcar_PRIVATE_DEPS)
+  list(REMOVE_DUPLICATES YARP_yarpcar_PRIVATE_DEPS)
+endif()
 yarp_install_basic_package_files(YARP_yarpcar
                                  FIRST_TARGET yarpcar
                                  DEPENDENCIES ${YARP_yarpcar_PUBLIC_DEPS}

--- a/src/devices/CMakeLists.txt
+++ b/src/devices/CMakeLists.txt
@@ -90,7 +90,9 @@ install(TARGETS yarpmod
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 include(YarpInstallBasicPackageFiles)
-list(REMOVE_DUPLICATES YARP_yarpmod_PRIVATE_DEPS)
+if(YARP_yarpmod_PRIVATE_DEPS)
+  list(REMOVE_DUPLICATES YARP_yarpmod_PRIVATE_DEPS)
+endif()
 yarp_install_basic_package_files(YARP_yarpmod
                                  FIRST_TARGET yarpmod
                                  DEPENDENCIES ${YARP_yarpmod_PUBLIC_DEPS}

